nemo/collections/asr/data/audio_to_audio.py

```diff
@@ -0,0 +1,936 @@
+                    # single-channel signal
+                    pad = (0, batch_length - s_len)
+                elif b[signal].ndim == 2:
+                    # multi-channel signal
+                    pad = (0, 0, 0, batch_length - s_len)
+                else:
+                    raise RuntimeError(
+                        f'Signal {signal} has unsuported dimensions {signal.shape}. Currently, only 1D and 2D arrays are supported.'
+                    )
+                b[signal] = torch.nn.functional.pad(b[signal], pad)
+            # append the current padded signal
+            b_signal.append(b[signal])
+        # (signal_batched, signal_length)
+        batched += (torch.stack(b_signal), torch.tensor(signal_length, dtype=torch.int32))
+
+    # Currently, outputs are expected to be in a tuple, where each element must correspond
+    # to the output type in the OrderedDict returned by output_types.
+    #
+    # Therefore, we return batched signals by interleaving signals and their length:
+    #   (signal_0, signal_0_length, signal_1, signal_1_length, ...)
+    return batched
+
+
+@experimental
+class BaseAudioDataset(Dataset):
+    """Base class of audio datasets, providing common functionality
+    for other audio datasets.
+
+    Each line of the manifest file is expected to have the following format
+        ```
+        {
+            audio_key[0]: 'path/to/audio_file_0',
+            audio_key[1]: 'path/to/audio_file_1',
+            ...
+            'duration': duration_of_input,
+        }
+        ```
+
+    Args:
+        manifest_filepath: Path to manifest file in a format described above.
+        sample_rate: Sample rate for loaded audio signals.
+        audio_to_manifest_key: Dictionary mapping audio signal labels to manifest keys.
+        audio_duration: Optional duration of each item returned by __getitem__.
+                        If `None`, complete audio will be loaded.
+                        If set, a subsegment will be loaded synchronously from
+                        target and audio, i.e., with the same start and end point.
+        random_offset: If `True`, offset will be randomized when loading a subsegment
+                       from a file.
+        max_duration: If audio exceeds this length, do not include in dataset.
+        min_duration: If audio is less than this length, do not include in dataset.
+        max_utts: Limit number of utterances.
+    """
+
+    @property
+    @abc.abstractmethod
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        """Returns definitions of module output ports.
+        """
+
+    def __init__(
+        self,
+        manifest_filepath: str,
+        sample_rate: int,
+        audio_to_manifest_key: Dict[str, Union[str, List[str]]],
+        audio_duration: Optional[float] = None,
+        random_offset: bool = False,
+        max_duration: Optional[float] = None,
+        min_duration: Optional[float] = None,
+        max_utts: int = 0,
+    ):
```

Comment:
Base  audio dataset will take object of that class as input and set it as variable. Note that this will NOT instantiate the class on its own, it must accept a subclasses value only. 

So basically if ASRAudioProcessor (tentative name) needs to be subclassed by a subclass of BaseAudioDataset then it will look like below

ASRAudioProcessor -> MySpecificAudioProcessor
BaseAudioProcessor __init__ -> MyCustomAudioToAudioDataset __init__ loads an object of MySpecificAudioProcessor and passes it to BaseAudioProcessor.__init)__